File: 2024/2024-12-07/fizzbuzz_test.rb
require 'minitest/autorun'
require './fizzbuzz.rb'

class FizzBuzzTest < Minitest::Test
    def test_fizzbuzz
        [
            {
                testcase: '3で割り切れるときはfizz',
                param: 3,
                expected: 'fizz'
            },
            {
                testcase: '5で割り切れるときはbuzz',
                param: 5,
                expected: 'buzz'
            },
            {
                testcase: '15で割り切れるときはfizzbuzz',
                param: 15,
                expected: 'fizzbuzz'
            },
            {
                testcase: '3,5,15で割り切れないときはそのまま出力する',
                param: 4,
                expected: '4'
            }
        ].each do |params|
            # Arrange
            param = params[:param]
            expected = params[:expected]
            # Act
            actual = fizzbuzz(param)

            assert_equal(expected, actual)
        end
    end
end

File: 2025/2025-01-25/src/__test__/fizzbuzz.test.ts
import { describe, expect, test } from '@jest/globals';
import { fizzbuzz } from '../fizzbuzz';

describe('fizzbuzz', () => {
    test('15の倍数のときを除いて3の倍数ならばfizz', () => {
        expect(fizzbuzz(3)).toBe('fizz');
        expect(fizzbuzz(15)).not.toBe('fizz');
    });

    test('15の倍数のときを除いて5の倍数ならばbuzz', () => {
        expect(fizzbuzz(5)).toBe('buzz');
        expect(fizzbuzz(15)).not.toBe('buzz');
    });

    test('3の倍数でも5の倍数でもないならば数値の文字列', () => {
        expect(fizzbuzz(12)).toBe('12');
    });
});